homeassistant/components/assist_pipeline/pipeline.py

```diff
@@ -394,6 +401,16 @@ def __post_init__(self) -> None:
             )
         pipeline_data.pipeline_runs[self.pipeline.id][self.id] = PipelineRunDebug()
 
+        # Directory to save audio for each pipeline run.
+        # Configured in YAML for assist_pipeline.
+        debug_recording_dir = self.hass.data.get(DATA_CONFIG, {}).get(
+            "debug_recording_dir"
+        )
+        if debug_recording_dir is not None:
+            self.debug_recording_dir = Path(debug_recording_dir) / self.id
+            _LOGGER.debug("Saving pipeline audio to %s", self.debug_recording_dir)
+            self.debug_recording_dir.mkdir(parents=True, exist_ok=True)
```

Comment:
Don't do I/O in the event loop. Wouldn't it be better to just do this when first pipeline is activated?